[BACKLOG-14619] Fix help URLs

AlexanderBuloichik opened the pull request.

wingman-pentaho commented:
### Build Completed

:fire: This pull request has some issues. It would be preferable to fix them in order for it to be just perfect. See below for more details. Some links are also available below for further assistance in addressing those issues.

#### Build Commands

```sh
ant -Dtestreports.dir=bin/reports/unit-test -f engine/build.xml clean-all resolve jacoco && ant -f engine/build.xml checkstyle publish-local
```

#### Cleanup Commands
```sh
rm -r ~/.ivy2/local || echo no publish local to remove
```

#### Changed files

```console
  engine/src/kettle-job-entries.xml
  engine/src/kettle-steps.xml
```

#### Unit test coverage change

These statistics help you identify how your changes have affected the coverage of the following files. If a file is not in this list, then its coverage was not affected by your changes. To get some help interpreting these metrics, please refer to [Jacoco's documentation](http://eclemma.org/jacoco/trunk/doc/counters.html).

###### org.pentaho.di.job.entries.ftpsget.JobEntryFTPSGet
 * Branch Change: **-.53%**:small_red_triangle_down:
 * Complexity Change: **-.56%**:small_red_triangle_down:
 * Instruction Change: **-.07%**:small_red_triangle_down:

###### org.pentaho.di.job.entries.getpop.MailConnectionMeta
 * Branch Change: **-1.00%**:small_red_triangle_down:
 * Complexity Change: **-1.41%**:small_red_triangle_down:
 * Instruction Change: **-.30%**:small_red_triangle_down:

###### org.pentaho.di.job.entries.waitforsql.JobEntryWaitForSQL
 * Branch Change: **-.66%**:small_red_triangle_down:
 * Complexity Change: **-.93%**:small_red_triangle_down:
 * Instruction Change: **-.13%**:small_red_triangle_down:

###### org.pentaho.di.trans.steps.combinationlookup.CombinationLookupMeta
 * Instruction Change:  + .04%

###### org.pentaho.di.trans.steps.datagrid.DataGridMeta
 * Instruction Change:  + .09%

###### org.pentaho.di.trans.steps.dbproc.DBProcMeta
 * Instruction Change:  + .11%

###### org.pentaho.di.trans.steps.dimensionlookup.DimensionLookupMeta
 * Branch Change:  + .31%
 * Instruction Change:  + .06%
 * Line Change: **-.23%**:small_red_triangle_down:
 * Method Change: **-1.02%**:small_red_triangle_down:

###### org.pentaho.di.trans.steps.groupby.GroupByMeta
 * Branch Change:  + 1.04%
 * Complexity Change:  + 1.05%
 * Instruction Change:  + .14%
 * Line Change:  + .37%

###### org.pentaho.di.trans.steps.ifnull.IfNullMeta
 * Branch Change: **-2.00%**:small_red_triangle_down:
 * Complexity Change: **-1.89%**:small_red_triangle_down:
 * Instruction Change: **-.17%**:small_red_triangle_down:

###### org.pentaho.di.trans.steps.insertupdate.InsertUpdateMeta
 * Instruction Change:  + .05%

###### org.pentaho.di.trans.steps.memgroupby.MemoryGroupByMeta
 * Branch Change:  + 1.18%
 * Complexity Change:  + 1.35%

###### org.pentaho.di.trans.steps.singlethreader.SingleThreaderMeta
 * Instruction Change:  + .10%

###### org.pentaho.di.trans.steps.sortedmerge.SortedMergeMeta
 * Instruction Change: **-.18%**:small_red_triangle_down:

###### org.pentaho.di.trans.steps.synchronizeaftermerge.SynchronizeAfterMergeMeta
 * Instruction Change: **-.04%**:small_red_triangle_down:

###### org.pentaho.di.trans.steps.textfileinput.TextFileInputMeta
 * Branch Change: **-.53%**:small_red_triangle_down:

###### org.pentaho.di.trans.steps.webservices.WebServiceMeta
 * Branch Change: **-1.79%**:small_red_triangle_down:
 * Complexity Change: **-1.19%**:small_red_triangle_down:
 * Instruction Change: **-.07%**:small_red_triangle_down:

mdamour1976 commented:
@AlexanderBuloichik for all help URL's that point to a "versioned" mindtouch link, you can just use the portion of the URL that identifies it, kettle will add the base url to that when we call getDocUrl.  The property that method uses needs to be updated to 7.1 if it hasn't already.

AlexanderBuloichik commented:
@mdamour1976 okay, will try

AlexanderBuloichik commented:
@mdamour1976 I added ability to use doc url without prefix in the xml files, and replaced "https://help.pentaho.com/Documentation/7.0/" prefix into "https://help.pentaho.com/Documentation/7.1/"

wingman-pentaho commented:
### Build Completed

:x: This pull request has errors. They will need to be addressed before it can be accepted. See below for more details. Some links are also available below for further assistance in addressing those issues.

#### Build Commands

```sh
ant -Dtestreports.dir=bin/reports/unit-test -f core/build.xml clean-all resolve jacoco && ant -f core/build.xml checkstyle publish-local
ant -Dtestreports.dir=bin/reports/unit-test -f engine/build.xml clean-all resolve jacoco && ant -f engine/build.xml checkstyle publish-local
```

#### Cleanup Commands
```sh
rm -r ~/.ivy2/local || echo no publish local to remove
```

#### Changed files

```console
  core/src/org/pentaho/di/core/messages/messages_en_US.properties
  core/src/org/pentaho/di/core/plugins/Plugin.java
  engine/src/kettle-job-entries.xml
  engine/src/kettle-steps.xml
```

#### Unit test coverage change

These statistics help you identify how your changes have affected the coverage of the following files. If a file is not in this list, then its coverage was not affected by your changes. To get some help interpreting these metrics, please refer to [Jacoco's documentation](http://eclemma.org/jacoco/trunk/doc/counters.html).

###### org.pentaho.di.core.Const
 * Branch Change:  + .20%
 * Complexity Change:  + .54%
 * Instruction Change:  + .38%
 * Line Change:  + .53%
 * Method Change:  + 1.69%

###### org.pentaho.di.core.plugins.Plugin
 * Instruction Change:  + .15%

###### org.pentaho.di.core.auth.core.impl.DefaultAuthenticationConsumerFactory
 * Branch Change: **-7.14%**:small_red_triangle_down:
 * Complexity Change: **-7.69%**:small_red_triangle_down:

###### org.pentaho.di.job.entries.evalfilesmetrics.JobEntryEvalFilesMetrics
 * Branch Change:  + .39%
 * Complexity Change:  + .51%
 * Instruction Change:  + .08%

###### org.pentaho.di.job.entries.evaluatetablecontent.JobEntryEvalTableContent
 * Branch Change: **-1.01%**:small_red_triangle_down:
 * Complexity Change: **-1.06%**:small_red_triangle_down:
 * Instruction Change: **-.17%**:small_red_triangle_down:

###### org.pentaho.di.job.entries.getpop.MailConnectionMeta
 * Branch Change:  + 1.00%
 * Complexity Change:  + 1.41%
 * Instruction Change:  + .30%

###### org.pentaho.di.trans.step.StepMeta
 * Instruction Change: **-.19%**:small_red_triangle_down:
 * Line Change: **-.26%**:small_red_triangle_down:

###### org.pentaho.di.trans.steps.calculator.CalculatorMetaFunction
 * Branch Change:  + 8.70%
 * Complexity Change:  + 3.08%
 * Instruction Change:  + .75%
 * Line Change:  + .89%

###### org.pentaho.di.trans.steps.combinationlookup.CombinationLookupMeta
 * Instruction Change: **-.04%**:small_red_triangle_down:

###### org.pentaho.di.trans.steps.groupby.GroupByMeta
 * Branch Change: **-1.04%**:small_red_triangle_down:
 * Complexity Change: **-1.05%**:small_red_triangle_down:
 * Instruction Change: **-.14%**:small_red_triangle_down:
 * Line Change: **-.37%**:small_red_triangle_down:

###### org.pentaho.di.trans.steps.insertupdate.InsertUpdateMeta
 * Instruction Change:  + .05%

###### org.pentaho.di.trans.steps.replacestring.ReplaceStringMeta
 * Instruction Change: **-.07%**:small_red_triangle_down:

###### org.pentaho.di.trans.steps.sortedmerge.SortedMergeMeta
 * Instruction Change:  + .18%

###### org.pentaho.di.trans.steps.synchronizeaftermerge.SynchronizeAfterMergeMeta
 * Instruction Change: **-.04%**:small_red_triangle_down:

###### org.pentaho.di.trans.steps.tableoutput.TableOutputMeta
 * Branch Change: **-.91%**:small_red_triangle_down:
 * Complexity Change: **-.82%**:small_red_triangle_down:

###### org.pentaho.di.trans.steps.textfileinput.TextFileInputMeta
 * Branch Change:  + .53%

###### org.pentaho.di.trans.steps.webservices.WebServiceMeta
 * Branch Change: **-1.79%**:small_red_triangle_down:
 * Complexity Change: **-1.19%**:small_red_triangle_down:
 * Instruction Change: **-.07%**:small_red_triangle_down:

#### License header violations

> :heavy_exclamation_mark: Copyright year is not the current year on core/src/org/pentaho/di/core/plugins/Plugin.java. Found 2013. Was expecting 2017

wingman-pentaho commented:
### Build Completed

:fire: This pull request has some issues. It would be preferable to fix them in order for it to be just perfect. See below for more details. Some links are also available below for further assistance in addressing those issues.

#### Build Commands

```sh
ant -Dtestreports.dir=bin/reports/unit-test -f core/build.xml clean-all resolve jacoco && ant -f core/build.xml checkstyle publish-local
ant -Dtestreports.dir=bin/reports/unit-test -f engine/build.xml clean-all resolve jacoco && ant -f engine/build.xml checkstyle publish-local
```

#### Cleanup Commands
```sh
rm -r ~/.ivy2/local || echo no publish local to remove
```

#### Changed files

```console
  core/src/org/pentaho/di/core/messages/messages_en_US.properties
  core/src/org/pentaho/di/core/plugins/Plugin.java
  engine/src/kettle-job-entries.xml
  engine/src/kettle-steps.xml
```

#### Unit test coverage change

These statistics help you identify how your changes have affected the coverage of the following files. If a file is not in this list, then its coverage was not affected by your changes. To get some help interpreting these metrics, please refer to [Jacoco's documentation](http://eclemma.org/jacoco/trunk/doc/counters.html).

###### org.pentaho.di.core.Const
 * Branch Change:  + .20%
 * Complexity Change:  + .54%
 * Instruction Change:  + .38%
 * Line Change:  + .53%
 * Method Change:  + 1.69%

###### org.pentaho.di.core.plugins.Plugin
 * Instruction Change:  + .15%

###### org.pentaho.di.core.auth.core.impl.DefaultAuthenticationConsumerFactory
 * Branch Change:  + 7.14%
 * Complexity Change:  + 7.69%

###### org.pentaho.di.job.entries.evalfilesmetrics.JobEntryEvalFilesMetrics
 * Branch Change:  + .39%
 * Complexity Change:  + .51%
 * Instruction Change:  + .08%

###### org.pentaho.di.job.entries.sql.JobEntrySQL
 * Instruction Change: **-3.28%**:small_red_triangle_down:
 * Line Change: **-3.87%**:small_red_triangle_down:

###### org.pentaho.di.trans.step.StepMeta
 * Instruction Change:  + .19%
 * Line Change:  + .26%

###### org.pentaho.di.trans.steps.checksum.CheckSumMeta
 * Instruction Change: **-.11%**:small_red_triangle_down:

###### org.pentaho.di.trans.steps.combinationlookup.CombinationLookupMeta
 * Instruction Change: **-.04%**:small_red_triangle_down:

###### org.pentaho.di.trans.steps.csvinput.CsvInputMeta
 * Instruction Change: **-.05%**:small_red_triangle_down:

###### org.pentaho.di.trans.steps.dimensionlookup.DimensionLookupMeta
 * Branch Change: **-.31%**:small_red_triangle_down:
 * Instruction Change: **-.06%**:small_red_triangle_down:
 * Line Change:  + .23%
 * Method Change:  + 1.02%

###### org.pentaho.di.trans.steps.groupby.GroupByMeta
 * Branch Change: **-1.04%**:small_red_triangle_down:
 * Complexity Change: **-1.05%**:small_red_triangle_down:
 * Instruction Change: **-.14%**:small_red_triangle_down:
 * Line Change: **-.37%**:small_red_triangle_down:

###### org.pentaho.di.trans.steps.ifnull.IfNullMeta
 * Branch Change: **-2.00%**:small_red_triangle_down:
 * Complexity Change: **-1.89%**:small_red_triangle_down:
 * Instruction Change: **-.17%**:small_red_triangle_down:

###### org.pentaho.di.trans.steps.insertupdate.InsertUpdateMeta
 * Instruction Change: **-.05%**:small_red_triangle_down:

###### org.pentaho.di.trans.steps.ldifinput.LDIFInputField
 * Branch Change:  + 4.17%
 * Complexity Change:  + 2.00%
 * Instruction Change:  + .38%

###### org.pentaho.di.trans.steps.memgroupby.MemoryGroupByMeta
 * Branch Change:  + 3.53%
 * Complexity Change:  + 4.05%

###### org.pentaho.di.trans.steps.rowgenerator.RowGeneratorMeta
 * Instruction Change:  + .08%

###### org.pentaho.di.trans.steps.sort.SortRowsMeta
 * Instruction Change:  + .08%

###### org.pentaho.di.trans.steps.synchronizeaftermerge.SynchronizeAfterMergeMeta
 * Instruction Change: **-.04%**:small_red_triangle_down:

###### org.pentaho.di.trans.steps.uniquerows.UniqueRowsMeta
 * Instruction Change:  + .18%
